### PR TITLE
adds tracking snippet to layout

### PR DESCRIPTION
**Pull request by paulmsmith**
image

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -31,5 +31,18 @@
     </main>
   </div>
   <%= javascript_pack_tag 'application', 'data-turbolinks-suppress-warning':'true', 'data-turbolinks-eval': "false", 'data-turbolinks-track': 'reload' %>
+  <% if Rails.env.production? %>
+    <%# Hotjar tracking tag  %>
+    <script>
```
Comment (gidsg) [inline app/views/layouts/application.html.erb]: Can we put `defer` attribute on this so it is non blocking?

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -31,5 +31,18 @@
...
+        (function(h,o,t,j,a,r){
+            h.hj=h.hj||function(){(h.hj.q=h.hj.q||[]).push(arguments)};
+            h._hjSettings={hjid:<%= Rails.configuration.x.hotjar.id %>,hjsv:<%= Rails.configuration.x.hotjar.snippet_version %>};
+            a=o.getElementsByTagName('head')[0];
+            r=o.createElement('script');r.async=1;
+            r.src=t+h._hjSettings.hjid+j+h._hjSettings.hjsv;
+            a.appendChild(r);
+        })(window,document,'https://static.hotjar.com/c/hotjar-','.js?sv=');
+    </script>
+  <% end %>
```
Comment (gidsg) [inline app/views/layouts/application.html.erb]: Can we move this into a partial and then conditionally include it?

File: config/environments/production.rb [hunk 3]
```diff
@@ -125,5 +125,7 @@
   config.x.salesforce.client_secret = ENV.fetch("SALESFORCE_CLIENT_SECRET")
   # TODO: update this with production host once credentials available
   config.x.salesforce.host = "test.salesforce.com"
+  config.x.hotjar.id = ENV.fetch("HOTJAR_ID")
+  config.x.hotjar.snippet_version = ENV.fetch("HOTJAR_SNIPPET_VERSION")
```
Comment (gidsg) [inline config/environments/production.rb]: I would just hardcode this rather than including it in environment as it's not a secret.

Comment (stuartmccoll) [reply config/environments/production.rb]: I was going with it here for ease of changing it - seemed like it'd be quicker for us to make an environment variable change if necessary than pushing up a code change.

Comment (gidsg) [reply config/environments/production.rb]: Not sure that's true as we need to do a deploy anyway to change an env var without downtime, so my preference if it's not a secret (which this isn't as it's exposed on the client-side), would be to hard-code it in config.

Comment (stuartmccoll) [reply config/environments/production.rb]: Ah! I was thinking about this yesterday and it seems like it's something we're not likely to need to change anyway - once we've got a Hotter key, I doubt we're likely to invalidate it and need another one. I'll strip this out and hardcode it in a partial.

File: app/views/layouts/application.html.erb (hunk 2 above)
Comment (paulmsmith) [reply app/views/layouts/application.html.erb]: We can do. It needs to be on every page so I just added it straight to layout. :)

File: config/environments/production.rb (hunk 3 above)
Comment (gidsg) [reply config/environments/production.rb]: @stuartmccoll doesn't like this change was pushed, can you push it up then I'll approve the PR?

Comment (stuartmccoll) [reply config/environments/production.rb]: I haven't done it yet - it's on my list next 👍.

Comment (paulmsmith) [reply config/environments/production.rb]: I've created the partial now @stuartmccoll if you pull down the latest from this branch

**Pull request merged**